src/main/java/org/sonarlint/intellij/core/ProjectBindingManager.java

```diff
@@ -19,23 +19,32 @@
  */
 package org.sonarlint.intellij.core;
 
+import com.intellij.concurrency.JobScheduler;
 import com.intellij.openapi.components.AbstractProjectComponent;
 import com.intellij.openapi.project.Project;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
 import org.sonarlint.intellij.config.global.SonarLintGlobalSettings;
 import org.sonarlint.intellij.config.global.SonarQubeServer;
 import org.sonarlint.intellij.config.project.SonarLintProjectSettings;
 import org.sonarlint.intellij.ui.SonarLintConsole;
+import org.sonarlint.intellij.util.SonarLintUtils;
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedSonarLintEngine;
+import org.sonarsource.sonarlint.core.client.api.connected.ServerConfiguration;
+import org.sonarsource.sonarlint.core.client.api.connected.StorageUpdateCheckResult;
 
 public class ProjectBindingManager extends AbstractProjectComponent {
   private final SonarLintEngineManager engineManager;
   private final SonarLintProjectSettings projectSettings;
   private final SonarLintGlobalSettings globalSettings;
   private final SonarLintProjectNotifications notifications;
   private final SonarLintConsole console;
+  private boolean hasGlobalUpdates;
+  private boolean hasUpdates;
```

Comment:
Initially I wanted to display that somewhere in the UI (so it was a kind of caching). But since it is in the event log, I think it can indeed be removed.